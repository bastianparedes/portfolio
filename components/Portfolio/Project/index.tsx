import React from 'react'
import { VscSignOut, VscGithub } from 'react-icons/vsc'
import Image from 'next/image'
import styles from './styles.module.css'
import stylesRight from './stylesRight.module.css'
import stylesLeft from './stylesLeft.module.css'

interface ProjectProps {
  index: number
  name: string
  link: string
  image: string
  github: string
  description: string
  technologies: string[]
}

const Project = ({
  index,
  name,
  link,
  image,
  github,
  description,
  technologies
}: ProjectProps): JSX.Element => {
  const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft
  const imagePath = '/images/projects/' + image

  return (
    <div className={styles.projectCard}>
      {link !== '' ? (
        <a
          className={`${styles.imageContainer} ${stylesSide.imageContainer}`}
          href={link}
          target="_blank"
          rel="noreferrer noopener"
        >
          <Image
            src={imagePath}
            alt={image}
            layout="fill"
            objectFit="contain"
          />
        </a>
      ) : (
        <div
          className={`${styles.imageContainer} ${stylesSide.imageContainer}`}
        >
          <Image
            src={imagePath}
            alt={image}
            layout="fill"
            objectFit="contain"
          />
        </div>
      )}

      <h2 className={`${styles.projectName} ${stylesSide.projectName}`}>
        {name}
      </h2>
      <div
        className={`${styles.projectDescriptionContainer} ${stylesSide.projectDescriptionContainer}`}
      >
        <p
          className={`${styles.projectDescription} ${stylesSide.projectDescription}`}
        >
          {description}
        </p>
      </div>
      <span
        className={`${styles.projectTechnologies} ${stylesSide.projectTechnologies}`}
      >
        {technologies.join(', ')}
      </span>
      <div className={`${styles.linksContainer} ${stylesSide.linksContainer}`}>
        {link !== '' ? (
          <a
            className={`${styles.linkWebsite} ${stylesSide.linkWebsite}`}
            href={link}
            target="_blank"
            rel="noreferrer noopener"
          >
            <VscSignOut />
          </a>
        ) : (
          ''
        )}
        {github !== '' ? (
          <a
            className={`${styles.linkGithub} ${stylesSide.linkGithub}`}
            href={github}
            target="_blank"
            rel="noreferrer noopener"
          >
            <VscGithub />
          </a>
        ) : (
          ''
        )}
      </div>
    </div>
  )
}

export default Project
