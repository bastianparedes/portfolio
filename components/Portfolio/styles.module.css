.projectsContainer {
  width: max(60%, 700px);
  display: grid;
  justify-content: center;
  align-items: center;
  gap: 10rem;
  padding-bottom: 10rem;
  margin: auto;
}
