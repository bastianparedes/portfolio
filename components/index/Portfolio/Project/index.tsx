import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/legacy/image';
import { VscGithub, VscSignOut } from 'react-icons/vsc';

import styles from './styles.module.scss';
import stylesLeft from './stylesLeft.module.scss';
import stylesRight from './stylesRight.module.scss';

interface ProjectProps {
  index: number;
  name: string;
  link?: string;
  image?: string;
  github?: string;
  description: string;
  technologies: string[];
}

const Project = ({
  index,
  name,
  link,
  image,
  github,
  description,
  technologies
}: ProjectProps): JSX.Element => {
  const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft;

  return (
    <div className={styles.projectCard}>
      {link !== undefined ? (
        <a
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
          href={link}
          rel="noreferrer noopener"
          target="_blank"
        >
          {image !== undefined && (
            <Image
              alt={image}
              layout="fill"
              objectFit="contain"
              src={'/images/projects/' + image}
            />
          )}
        </a>
      ) : (
        <div
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
        >
          {image !== undefined && (
            <Image
              alt={image}
              layout="fill"
              objectFit="contain"
              src={'/images/projects/' + image}
            />
          )}
        </div>
      )}

      <h2 className={classNames(styles.projectName, stylesSide.projectName)}>
        {name}
      </h2>
      <div
        className={classNames(
          styles.projectDescriptionContainer,
          stylesSide.projectDescriptionContainer
        )}
      >
        <p className={styles.projectDescription}>{description}</p>
      </div>
      <span
        className={classNames(
          styles.projectTechnologies,
          stylesSide.projectTechnologies
        )}
      >
        {technologies.join(', ')}
      </span>
      <div
        className={classNames(styles.linksContainer, stylesSide.linksContainer)}
      >
        {link !== undefined && (
          <a
            className={styles.linkWebsite}
            href={link}
            rel="noreferrer noopener"
            target="_blank"
          >
            <VscSignOut />
          </a>
        )}
        {github !== undefined && (
          <a
            className={styles.linkGithub}
            href={github}
            rel="noreferrer noopener"
            target="_blank"
          >
            <VscGithub />
          </a>
        )}
      </div>
    </div>
  );
};

export default Project;
