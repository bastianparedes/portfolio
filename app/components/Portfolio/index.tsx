import React from 'react';

import Project from './Project';
import styles from './styles.module.scss';
import constants from '../../../config/constants';
import { Section } from '../Common';

const Portfolio = (): JSX.Element => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {constants.PORTFOLIO.PROJECTS.map((info, index) => (
          <Project
            description={info.DESCRIPTION}
            github={info.GITHUB}
            image={info.IMAGE}
            index={index}
            key={info.NAME}
            link={info.LINK}
            name={info.NAME}
            technologies={info.TECHNOLOGIES}
          />
        ))}
      </div>
    </Section>
  );
};

export default Portfolio;
