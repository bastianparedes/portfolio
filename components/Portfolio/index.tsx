import React from 'react'
import { Section } from '../Common'
import styles from './styles.module.css'
import Project from './Project'
import constants from '../../config/constants.json'

const Portfolio = (): JSX.Element => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {constants.PORTFOLIO.PROJECTS.map((info, index) => (
          <Project
            key={info.NAME}
            index={index}
            name={info.NAME}
            link={info.LINK}
            image={info.IMAGE}
            github={info.GITHUB}
            description={info.DESCRIPTION}
            technologies={info.TECHNOLOGIES}
          />
        ))}
      </div>
    </Section>
  )
}

export default Portfolio
