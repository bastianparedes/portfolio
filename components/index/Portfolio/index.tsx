import React from 'react';

import constants from '../../../config/constants.json';
import { Section } from '../Common';
import Project from './Project';
import styles from './styles.module.scss';

const Portfolio = (): JSX.Element => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {constants.PORTFOLIO.PROJECTS.map((info, index) => (
          <Project
            key={info.NAME}
            index={index}
            name={info.NAME}
            link={info.LINK}
            image={info.IMAGE}
            github={info.GITHUB}
            description={info.DESCRIPTION}
            technologies={info.TECHNOLOGIES}
          />
        ))}
      </div>
    </Section>
  );
};

export default Portfolio;
