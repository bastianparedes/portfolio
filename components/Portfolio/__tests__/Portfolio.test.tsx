import React from 'react'
import Portfolio from '..'
import { render } from '@testing-library/react'

describe('<Portfolio />', () => {
  it('should render', () => {
    const { container } = render(<Portfolio />)
    expect(container).toMatchSnapshot()
  })
})
