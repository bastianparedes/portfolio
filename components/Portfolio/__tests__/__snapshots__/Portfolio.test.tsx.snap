// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> should render 1`] = `
<div>
  <section
    class="section"
    id="portfolio"
  >
    <h3
      class="title"
    >
      Portfolio
    </h3>
    <div
      class="projectsContainer"
    >
      <div
        class="projectCard"
      >
        <a
          class="imageContainer imageContainer"
          href="https://www.falabella.com/falabella-cl/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="falabella.png"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
            <noscript />
          </span>
        </a>
        <h2
          class="projectName projectName"
        >
          Falabella
        </h2>
        <div
          class="projectDescriptionContainer projectDescriptionContainer"
        >
          <p
            class="projectDescription projectDescription"
          >
            A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
          </p>
        </div>
        <span
          class="projectTechnologies projectTechnologies"
        >
          HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
        </span>
        <div
          class="linksContainer linksContainer"
        >
          <a
            class="linkWebsite linkWebsite"
            href="https://www.falabella.com/falabella-cl/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.02 3.77v1.56l1-.99V2.5l-.5-.5h-9l-.5.5v.486L2 3v10.29l.36.46 5 1.72L8 15v-1h3.52l.5-.5v-1.81l-1-1V13H8V4.71l-.33-.46L4.036 3h6.984v.77zM7 14.28l-4-1.34V3.72l4 1.34v9.22zm6.52-5.8H8.55v-1h4.93l-1.6-1.6.71-.7 2.47 2.46v.71l-2.49 2.48-.7-.7 1.65-1.65z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="projectCard"
      >
        <a
          class="imageContainer imageContainer"
          href="https://tusalad.vercel.app/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="tu-salad.png"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
            <noscript />
          </span>
        </a>
        <h2
          class="projectName projectName"
        >
          Tu Salad
        </h2>
        <div
          class="projectDescriptionContainer projectDescriptionContainer"
        >
          <p
            class="projectDescription projectDescription"
          >
            Página web de venta de bowls de ensalada que incluye un carrito de compras y una pasarela de pago construida mediante el servicio Webpay plus de Transbank. Las compras quedan registradas en una base de datos para consultar su estado.
          </p>
        </div>
        <span
          class="projectTechnologies projectTechnologies"
        >
          React.js, Next.js, TypeScript, WebPay
        </span>
        <div
          class="linksContainer linksContainer"
        >
          <a
            class="linkWebsite linkWebsite"
            href="https://tusalad.vercel.app/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.02 3.77v1.56l1-.99V2.5l-.5-.5h-9l-.5.5v.486L2 3v10.29l.36.46 5 1.72L8 15v-1h3.52l.5-.5v-1.81l-1-1V13H8V4.71l-.33-.46L4.036 3h6.984v.77zM7 14.28l-4-1.34V3.72l4 1.34v9.22zm6.52-5.8H8.55v-1h4.93l-1.6-1.6.71-.7 2.47 2.46v.71l-2.49 2.48-.7-.7 1.65-1.65z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
          <a
            class="linkGithub linkGithub"
            href="https://github.com/BastianParedes/tusalad/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm3.163 21.783h-.093a.513.513 0 0 1-.382-.14.513.513 0 0 1-.14-.372v-1.406c.006-.467.01-.94.01-1.416a3.693 3.693 0 0 0-.151-1.028 1.832 1.832 0 0 0-.542-.875 8.014 8.014 0 0 0 2.038-.471 4.051 4.051 0 0 0 1.466-.964c.407-.427.71-.943.885-1.506a6.77 6.77 0 0 0 .3-2.13 4.138 4.138 0 0 0-.26-1.476 3.892 3.892 0 0 0-.795-1.284 2.81 2.81 0 0 0 .162-.582c.033-.2.05-.402.05-.604 0-.26-.03-.52-.09-.773a5.309 5.309 0 0 0-.221-.763.293.293 0 0 0-.111-.02h-.11c-.23.002-.456.04-.674.111a5.34 5.34 0 0 0-.703.26 6.503 6.503 0 0 0-.661.343c-.215.127-.405.249-.573.362a9.578 9.578 0 0 0-5.143 0 13.507 13.507 0 0 0-.572-.362 6.022 6.022 0 0 0-.672-.342 4.516 4.516 0 0 0-.705-.261 2.203 2.203 0 0 0-.662-.111h-.11a.29.29 0 0 0-.11.02 5.844 5.844 0 0 0-.23.763c-.054.254-.08.513-.081.773 0 .202.017.404.051.604.033.199.086.394.16.582A3.888 3.888 0 0 0 5.702 10a4.142 4.142 0 0 0-.263 1.476 6.871 6.871 0 0 0 .292 2.12c.181.563.483 1.08.884 1.516.415.422.915.75 1.466.964.653.25 1.337.41 2.033.476a1.828 1.828 0 0 0-.452.633 2.99 2.99 0 0 0-.2.744 2.754 2.754 0 0 1-1.175.27 1.788 1.788 0 0 1-1.065-.3 2.904 2.904 0 0 1-.752-.824 3.1 3.1 0 0 0-.292-.382 2.693 2.693 0 0 0-.372-.343 1.841 1.841 0 0 0-.432-.24 1.2 1.2 0 0 0-.481-.101c-.04.001-.08.005-.12.01a.649.649 0 0 0-.162.02.408.408 0 0 0-.13.06.116.116 0 0 0-.06.1.33.33 0 0 0 .14.242c.093.074.17.131.232.171l.03.021c.133.103.261.214.382.333.112.098.213.209.3.33.09.119.168.246.231.381.073.134.15.288.231.463.188.474.522.875.954 1.145.453.243.961.364 1.476.351.174 0 .349-.01.522-.03.172-.028.343-.057.515-.091v1.743a.5.5 0 0 1-.533.521h-.062a10.286 10.286 0 1 1 6.324 0v.005z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="projectCard"
      >
        <a
          class="imageContainer imageContainer"
          href="https://pdf-constructor.vercel.app/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="pdf-constructor.png"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
            <noscript />
          </span>
        </a>
        <h2
          class="projectName projectName"
        >
          PDF Constructor
        </h2>
        <div
          class="projectDescriptionContainer projectDescriptionContainer"
        >
          <p
            class="projectDescription projectDescription"
          >
            Es una página web que recibe las imágenes cargadas por el usuario, las configura a gusto y finalmente genera un archivo PDF que se guarda en el computador del mismo. Todo el proceso ocurre del lado del Front-end.
          </p>
        </div>
        <span
          class="projectTechnologies projectTechnologies"
        >
          React.js, SortableJS, jsPDF
        </span>
        <div
          class="linksContainer linksContainer"
        >
          <a
            class="linkWebsite linkWebsite"
            href="https://pdf-constructor.vercel.app/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.02 3.77v1.56l1-.99V2.5l-.5-.5h-9l-.5.5v.486L2 3v10.29l.36.46 5 1.72L8 15v-1h3.52l.5-.5v-1.81l-1-1V13H8V4.71l-.33-.46L4.036 3h6.984v.77zM7 14.28l-4-1.34V3.72l4 1.34v9.22zm6.52-5.8H8.55v-1h4.93l-1.6-1.6.71-.7 2.47 2.46v.71l-2.49 2.48-.7-.7 1.65-1.65z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
          <a
            class="linkGithub linkGithub"
            href="https://github.com/BastianParedes/pdf-constructor/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm3.163 21.783h-.093a.513.513 0 0 1-.382-.14.513.513 0 0 1-.14-.372v-1.406c.006-.467.01-.94.01-1.416a3.693 3.693 0 0 0-.151-1.028 1.832 1.832 0 0 0-.542-.875 8.014 8.014 0 0 0 2.038-.471 4.051 4.051 0 0 0 1.466-.964c.407-.427.71-.943.885-1.506a6.77 6.77 0 0 0 .3-2.13 4.138 4.138 0 0 0-.26-1.476 3.892 3.892 0 0 0-.795-1.284 2.81 2.81 0 0 0 .162-.582c.033-.2.05-.402.05-.604 0-.26-.03-.52-.09-.773a5.309 5.309 0 0 0-.221-.763.293.293 0 0 0-.111-.02h-.11c-.23.002-.456.04-.674.111a5.34 5.34 0 0 0-.703.26 6.503 6.503 0 0 0-.661.343c-.215.127-.405.249-.573.362a9.578 9.578 0 0 0-5.143 0 13.507 13.507 0 0 0-.572-.362 6.022 6.022 0 0 0-.672-.342 4.516 4.516 0 0 0-.705-.261 2.203 2.203 0 0 0-.662-.111h-.11a.29.29 0 0 0-.11.02 5.844 5.844 0 0 0-.23.763c-.054.254-.08.513-.081.773 0 .202.017.404.051.604.033.199.086.394.16.582A3.888 3.888 0 0 0 5.702 10a4.142 4.142 0 0 0-.263 1.476 6.871 6.871 0 0 0 .292 2.12c.181.563.483 1.08.884 1.516.415.422.915.75 1.466.964.653.25 1.337.41 2.033.476a1.828 1.828 0 0 0-.452.633 2.99 2.99 0 0 0-.2.744 2.754 2.754 0 0 1-1.175.27 1.788 1.788 0 0 1-1.065-.3 2.904 2.904 0 0 1-.752-.824 3.1 3.1 0 0 0-.292-.382 2.693 2.693 0 0 0-.372-.343 1.841 1.841 0 0 0-.432-.24 1.2 1.2 0 0 0-.481-.101c-.04.001-.08.005-.12.01a.649.649 0 0 0-.162.02.408.408 0 0 0-.13.06.116.116 0 0 0-.06.1.33.33 0 0 0 .14.242c.093.074.17.131.232.171l.03.021c.133.103.261.214.382.333.112.098.213.209.3.33.09.119.168.246.231.381.073.134.15.288.231.463.188.474.522.875.954 1.145.453.243.961.364 1.476.351.174 0 .349-.01.522-.03.172-.028.343-.057.515-.091v1.743a.5.5 0 0 1-.533.521h-.062a10.286 10.286 0 1 1 6.324 0v.005z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="projectCard"
      >
        <a
          class="imageContainer imageContainer"
          href="https://my-global-chat.vercel.app/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="my-global-chat.png"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
            <noscript />
          </span>
        </a>
        <h2
          class="projectName projectName"
        >
          My Global Chat
        </h2>
        <div
          class="projectDescriptionContainer projectDescriptionContainer"
        >
          <p
            class="projectDescription projectDescription"
          >
            Aplicación web que pide al usuario un nombre y lo lleva a una sala donde todos los conectados a la página en ese momento pueden conversar al mismo tiempo.
          </p>
        </div>
        <span
          class="projectTechnologies projectTechnologies"
        >
          React.js, Socket.IO
        </span>
        <div
          class="linksContainer linksContainer"
        >
          <a
            class="linkWebsite linkWebsite"
            href="https://my-global-chat.vercel.app/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.02 3.77v1.56l1-.99V2.5l-.5-.5h-9l-.5.5v.486L2 3v10.29l.36.46 5 1.72L8 15v-1h3.52l.5-.5v-1.81l-1-1V13H8V4.71l-.33-.46L4.036 3h6.984v.77zM7 14.28l-4-1.34V3.72l4 1.34v9.22zm6.52-5.8H8.55v-1h4.93l-1.6-1.6.71-.7 2.47 2.46v.71l-2.49 2.48-.7-.7 1.65-1.65z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
          <a
            class="linkGithub linkGithub"
            href="https://github.com/BastianParedes/my-global-chat/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm3.163 21.783h-.093a.513.513 0 0 1-.382-.14.513.513 0 0 1-.14-.372v-1.406c.006-.467.01-.94.01-1.416a3.693 3.693 0 0 0-.151-1.028 1.832 1.832 0 0 0-.542-.875 8.014 8.014 0 0 0 2.038-.471 4.051 4.051 0 0 0 1.466-.964c.407-.427.71-.943.885-1.506a6.77 6.77 0 0 0 .3-2.13 4.138 4.138 0 0 0-.26-1.476 3.892 3.892 0 0 0-.795-1.284 2.81 2.81 0 0 0 .162-.582c.033-.2.05-.402.05-.604 0-.26-.03-.52-.09-.773a5.309 5.309 0 0 0-.221-.763.293.293 0 0 0-.111-.02h-.11c-.23.002-.456.04-.674.111a5.34 5.34 0 0 0-.703.26 6.503 6.503 0 0 0-.661.343c-.215.127-.405.249-.573.362a9.578 9.578 0 0 0-5.143 0 13.507 13.507 0 0 0-.572-.362 6.022 6.022 0 0 0-.672-.342 4.516 4.516 0 0 0-.705-.261 2.203 2.203 0 0 0-.662-.111h-.11a.29.29 0 0 0-.11.02 5.844 5.844 0 0 0-.23.763c-.054.254-.08.513-.081.773 0 .202.017.404.051.604.033.199.086.394.16.582A3.888 3.888 0 0 0 5.702 10a4.142 4.142 0 0 0-.263 1.476 6.871 6.871 0 0 0 .292 2.12c.181.563.483 1.08.884 1.516.415.422.915.75 1.466.964.653.25 1.337.41 2.033.476a1.828 1.828 0 0 0-.452.633 2.99 2.99 0 0 0-.2.744 2.754 2.754 0 0 1-1.175.27 1.788 1.788 0 0 1-1.065-.3 2.904 2.904 0 0 1-.752-.824 3.1 3.1 0 0 0-.292-.382 2.693 2.693 0 0 0-.372-.343 1.841 1.841 0 0 0-.432-.24 1.2 1.2 0 0 0-.481-.101c-.04.001-.08.005-.12.01a.649.649 0 0 0-.162.02.408.408 0 0 0-.13.06.116.116 0 0 0-.06.1.33.33 0 0 0 .14.242c.093.074.17.131.232.171l.03.021c.133.103.261.214.382.333.112.098.213.209.3.33.09.119.168.246.231.381.073.134.15.288.231.463.188.474.522.875.954 1.145.453.243.961.364 1.476.351.174 0 .349-.01.522-.03.172-.028.343-.057.515-.091v1.743a.5.5 0 0 1-.533.521h-.062a10.286 10.286 0 1 1 6.324 0v.005z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="projectCard"
      >
        <div
          class="imageContainer imageContainer"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="tstmaker.png"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
            <noscript />
          </span>
        </div>
        <h2
          class="projectName projectName"
        >
          TstMaker
        </h2>
        <div
          class="projectDescriptionContainer projectDescriptionContainer"
        >
          <p
            class="projectDescription projectDescription"
          >
            Aplicación de escritorio que construye evaluaciones de seleción única de matemática para el nivel secundario de las escuelas chilenas.
          </p>
        </div>
        <span
          class="projectTechnologies projectTechnologies"
        >
          Python3, Tkinter, MiKTeX
        </span>
        <div
          class="linksContainer linksContainer"
        >
          <a
            class="linkGithub linkGithub"
            href="https://github.com/BastianParedes/tstmaker/"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm3.163 21.783h-.093a.513.513 0 0 1-.382-.14.513.513 0 0 1-.14-.372v-1.406c.006-.467.01-.94.01-1.416a3.693 3.693 0 0 0-.151-1.028 1.832 1.832 0 0 0-.542-.875 8.014 8.014 0 0 0 2.038-.471 4.051 4.051 0 0 0 1.466-.964c.407-.427.71-.943.885-1.506a6.77 6.77 0 0 0 .3-2.13 4.138 4.138 0 0 0-.26-1.476 3.892 3.892 0 0 0-.795-1.284 2.81 2.81 0 0 0 .162-.582c.033-.2.05-.402.05-.604 0-.26-.03-.52-.09-.773a5.309 5.309 0 0 0-.221-.763.293.293 0 0 0-.111-.02h-.11c-.23.002-.456.04-.674.111a5.34 5.34 0 0 0-.703.26 6.503 6.503 0 0 0-.661.343c-.215.127-.405.249-.573.362a9.578 9.578 0 0 0-5.143 0 13.507 13.507 0 0 0-.572-.362 6.022 6.022 0 0 0-.672-.342 4.516 4.516 0 0 0-.705-.261 2.203 2.203 0 0 0-.662-.111h-.11a.29.29 0 0 0-.11.02 5.844 5.844 0 0 0-.23.763c-.054.254-.08.513-.081.773 0 .202.017.404.051.604.033.199.086.394.16.582A3.888 3.888 0 0 0 5.702 10a4.142 4.142 0 0 0-.263 1.476 6.871 6.871 0 0 0 .292 2.12c.181.563.483 1.08.884 1.516.415.422.915.75 1.466.964.653.25 1.337.41 2.033.476a1.828 1.828 0 0 0-.452.633 2.99 2.99 0 0 0-.2.744 2.754 2.754 0 0 1-1.175.27 1.788 1.788 0 0 1-1.065-.3 2.904 2.904 0 0 1-.752-.824 3.1 3.1 0 0 0-.292-.382 2.693 2.693 0 0 0-.372-.343 1.841 1.841 0 0 0-.432-.24 1.2 1.2 0 0 0-.481-.101c-.04.001-.08.005-.12.01a.649.649 0 0 0-.162.02.408.408 0 0 0-.13.06.116.116 0 0 0-.06.1.33.33 0 0 0 .14.242c.093.074.17.131.232.171l.03.021c.133.103.261.214.382.333.112.098.213.209.3.33.09.119.168.246.231.381.073.134.15.288.231.463.188.474.522.875.954 1.145.453.243.961.364 1.476.351.174 0 .349-.01.522-.03.172-.028.343-.057.515-.091v1.743a.5.5 0 0 1-.533.521h-.062a10.286 10.286 0 1 1 6.324 0v.005z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
